### PR TITLE
[RFC] Rewrite GUI to use JavaFX

### PR DESCRIPTION
**Pull request by Unia**
From #30, it was concluded we needed to switch libraries. A result of this was that we were free to choose another GUI library, which unanimously became JavaFX. This PR is the result of that decision.

Should I test `DNAinator`, `WindowController`, `GraphController` and/or `ExceptionDialog`?

*NOTE:* this PR depends on #33, in that it already provides a test case for `FileLoadService` with a faulty edge file.

### CODE REVIEW

**Comment by pvanagtmaal** [comment]
Line 33 in ExceptionDialog, why would you want to call static methods on GridPane before creating a GridPane?

**Comment by Unia** [comment]
Good catch on the CTRL+O - the other shurtcuts should work as-is but I somehow missed it :)

Those methods are static because... well, they [are](https://docs.oracle.com/javase/8/javafx/api/javafx/scene/layout/GridPane.html#setVgrow-javafx.scene.Node-javafx.scene.layout.Priority-).

I will push an update to the Travis configuration that might resolve our build failure, too.

**Comment by Unia** [comment]
Updated Travis configuration to start a display for us: http://docs.travis-ci.com/user/gui-and-headless-browsers/#Using-xvfb-to-Run-Tests-That-Require-GUI-%28e.g.-a-Web-browser%29

**Comment by pvanagtmaal** [comment]
Looking good now. Waiting for a second reviewer before merging :)

File: src/main/java/nl/tudelft/dnainator/widgets/ExceptionDialog.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ * Provides a dialog to inform the user that an exception has occurred.
+ * <p>
+ * Based on code from a tutorial found here: http://code.makery.ch/blog/javafx-dialogs-official/
+ * </p>
+ */
+public class ExceptionDialog {
+
+	/**
+	 * Instantiates a new ExceptionDialog.
+	 *
```
Comment (ohuang12) [inline src/main/java/nl/tudelft/dnainator/widgets/ExceptionDialog.java]: Remove the extra line :)

File: src/main/java/nl/tudelft/dnainator/controllers/WindowController.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		chooser.getExtensionFilters().add(
+				new ExtensionFilter("Graph files", "*.node.graph"));
+		File nodeFile = chooser.showOpenDialog(root.getScene().getWindow());
+		if (nodeFile == null) {
+			return;
+		}
+		loadService.setNodeFile(nodeFile);
+		loadService.setEdgeFile(openEdgeFile(nodeFile.getPath()));
+		loadService.restart();
+    }
```
Comment (ohuang12) [inline src/main/java/nl/tudelft/dnainator/controllers/WindowController.java]: Odd indent?

**Comment by pvanagtmaal** [comment]
Merging!

**Pull request merged**